subsys/net/l2/ieee802154/ieee802154.c

```diff
@@ -453,12 +454,13 @@ static enum net_verdict ieee802154_recv(struct net_if *iface, struct net_pkt *pk
 
 #ifdef CONFIG_NET_6LO
 	verdict = ieee802154_6lo_decode_pkt(iface, pkt);
+#endif /* CONFIG_NET_6LO */
+
+	if (verdict == NET_CONTINUE) {
+		pkt_hexdump(RX_PKT_TITLE, pkt, true);
+	}
 
-	pkt_hexdump(RX_PKT_TITLE, pkt, true);
 	return verdict;
```

Comment:
How you can guarantee that you have a valid verdict here if you dropped the default value?